File: transpiler/javatests/com/google/j2cl/readable/java/bridgemethods/BridgeMethodShadowedSignature.java
/*
 * Copyright 2017 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package bridgemethods;

import java.util.function.Consumer;

public class BridgeMethodShadowedSignature {

  interface I<I1> {
    String get(Consumer<? super I1> consumer);
  }

  abstract static class B<B1, B2> implements I<B1> {
    public String get(B2 consumer) {
      return "B get B2";
    }
  }

  static class C<C1> extends B<C1, Consumer<? super C1>> implements I<C1> {
    // Needs a bridge from the I.get(Consumer) signature to B.get(Object) implementation.
  }
}

File: transpiler/javatests/com/google/j2cl/readable/kotlin/bridgemethods/BridgeMethodShadowedSignature.kt
/*
 * Copyright 2022 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package bridgemethods

import java.util.function.Consumer

class BridgeMethodShadowedSignature {
  internal interface I<I1> {
    fun get(consumer: Consumer<I1>): String?
  }

  abstract class B<B1, B2> : I<B1> {
    fun get(consumer: B2): String {
      return "B get B2"
    }
  }

  internal class C<C1> :
    B<C1, Consumer<C1>>(),
    I<C1> { // Needs a bridge from the I.get(Consumer) signature to B.get(Object) implementation.
  }
}
